go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1846,25 +1846,31 @@ func commonFlows(hostSubnets []*net.IPNet, bridge *bridgeConfiguration) ([]strin
 							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, config.Default.ConntrackZone,
 							netConfig.masqCTMark, ofPortPhys))
 					// IP traffic from the OVN network to the host network should be handled normally by the bridge instead of
-					// being forwarded directly to the NIC (prio=100 flow above). This allows pods in the default network to reach
-					// localnet pods on the same node.
+					// being forwarded directly to the NIC (prio=100 flow above). This allows:
+					// (a) default network pods -> localnet pod (same node)
+					// (b) localnet pod -> host (same node)
 					for _, hostSubnet := range hostSubnets {
 						if hostSubnet.IP.To4() == nil {
 							continue
 						}
-						newFlow := fmt.Sprintf("cookie=%s, priority=102, in_port=%s, dl_src=%s, ip, nw_dst=%s, "+
-							"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
-							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, hostSubnet,
-							config.Default.ConntrackZone, netConfig.masqCTMark)
-
-						if config.Gateway.Mode == config.GatewayModeLocal {
-							newFlow = fmt.Sprintf("cookie=%s, priority=102, in_port=LOCAL, dl_src=%s, ip, nw_dst=%s, "+
+						newFlows := []string{}
+						// scenario (a) for shared gw mode
+						if config.Gateway.Mode == config.GatewayModeShared {
+							newFlows = append(newFlows,
+								fmt.Sprintf("cookie=%s, priority=102, in_port=%s, dl_src=%s, ip, nw_dst=%s, "+
+									"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
+									defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, hostSubnet,
+									config.Default.ConntrackZone, netConfig.masqCTMark))
+						}
+						// scenario (a) for local gw mode
+						// scenario (b) for both gw modes
```

Comment:
maybe it's just me and my ignorance is getting in the way, but it's not immediately clear why for both GW modes we **need** this second flow.

Reply:
For localnet<->host, regardless of the gateway mode,  we need the same flow that allows OVN->localnet, with `in_port=LOCAL` and the host subnet as dest ip in the filter. I can try to rephrase it better in the comments.

Reply:
please do; thanks for explaining.